tests/framework/session/test_session_extension_hooks.py

```diff
@@ -562,6 +578,34 @@ def test_after_dataset_load_hook_async(
             ["Before dataset loaded", "Catalog load", "After dataset loaded"]
         ).strip("[]") in str(hooks_log_messages).strip("[]")
 
+    def test_after_dataset_load_hook_async_multiple_outputs(
+        self,
+        mocker,
+        memory_catalog,
+        mock_session,
+        sample_node_multiple_outputs,
+    ):
+        # load mock context to instantiate Hooks
+        mock_session.load_context()
+
+        after_dataset_saved_mock = mocker.patch.object(
+            mock_session._hook_manager.hook, "after_dataset_saved"
+        )
+        _run_node_async(
+            node=sample_node_multiple_outputs,
+            catalog=memory_catalog,
+            hook_manager=mock_session._hook_manager,
+        )
+
+        after_dataset_saved_mock.assert_has_calls(
```

Comment:
This looks like a nice test! Could you confirm that this test will fail if your change in `runner.py` is not implemented?

Reply:
Yes, I wrote this test on main (where it failed) before adding it to this branch 🙂.